matriculation-plugin/src/main/java/fi/otavanopisto/muikku/plugins/matriculation/MatriculationRESTService.java

```diff
@@ -0,0 +1,165 @@
+  
+  @GET
+  @RESTPermit(MatriculationPermissions.MATRICULATION_GET_EXAM)
+  @Path("/currentExam")
+  public Response fetchCurrentExam() {
+    MatriculationExam exam = matriculationSchoolDataBridge.getMatriculationExam();
+    return Response.ok(exam).build();
+  }
+  
+  @GET
+  @RESTPermit(MatriculationPermissions.MATRICULATION_GET_INITIALDATA)
+  @Path("/initialData/{USERID}")
+  public Response fetchInitialData(@PathParam("USERID") String userId) {
+    MatriculationExamInitialData result = new MatriculationExamInitialData();
+    SchoolDataIdentifier identifier = SchoolDataIdentifier.fromId(userId);
+    if (identifier == null) {
+      return Response.status(Status.BAD_REQUEST).entity("Invalid user id").build();
+    }
+    long studentId = Long.valueOf(identifier.getIdentifier().split("-")[1]);
+    User user = userController.findUserByIdentifier(identifier);
+    if (user == null) {
+      return Response.status(Status.NOT_FOUND).entity("User not found").build();
+    }
+    UserEntity userEntity = userEntityController.findUserEntityByUserIdentifier(identifier);
+    List<UserAddress> userAddresses = userController.listUserAddresses(user);
+    List<UserPhoneNumber> phoneNumbers = userController.listUserPhoneNumbers(user);
+    String address = "";
+    String postalCode = "";
+    String locality = "";
+    String phoneNumber = "";
+    String emailAddress = userEmailEntityController.getUserDefaultEmailAddress(userEntity, false); 
+    String name = "";
+    if (user.getFirstName() != null) {
+      name += user.getFirstName();
+    }
+    if (user.getLastName() != null) {
+      name += " " + user.getLastName();
+    }
+    for (UserAddress userAddress : userAddresses) {
+      if (userAddress.getDefaultAddress()) {
+        address = userAddress.getStreet();
+        postalCode = userAddress.getPostalCode();
+        locality = userAddress.getCity();
+      }
+    }
+    for (UserPhoneNumber userPhoneNumber : phoneNumbers) {
+      if (userPhoneNumber.getDefaultNumber()) {
+        phoneNumber = userPhoneNumber.getNumber();
+      }
+    }
+    result.setName(name);
+    result.setSsn(user.getSsn());
+    result.setEmail(emailAddress);
+    result.setPhone(phoneNumber);
+    result.setAddress(address);
+    result.setPostalCode(postalCode);
+    result.setLocality(locality);
+    result.setGuidanceCounselor("");
+    result.setStudentId(studentId);
+    
+    try {
+      VopsLister.Result listerResult = torController.listVopsCourses(
+        userId,
+        identifier);
+
+      result.setMandatoryCourses(listerResult.getNumMandatoryCourses());
+    } catch (EducationTypeMappingNotSetException ex) {
+      result.setMandatoryCourses(0);
+    }
+    
+    return Response.ok(result).build();
+  }
+  
+  @POST
+  @RESTPermit(MatriculationPermissions.MATRICULATION_SEND_ENROLLMENT)
+  @Path("/enrollments")
+  public Response sendEnrollment(MatriculationExamEnrollment enrollment) {
```

Comment:
Same as previous comment. When a student enrolls to the matriculation exam, the payload should be validated to ensure that it belongs to them.